TypeSystem/flow-analysis/reachability_do_while_A01_t02.dart

```diff
@@ -12,9 +12,42 @@
 /// definitely unassigned in `E`.
 /// @author sgrekhov22@gmail.com
 
-main() {
+class C {
+  int v;
+  C(this.v);
+}
+
+test1() {
   late int i;
   do {
     i = 42;
   } while (i < 0);
 }
+
+test2() {
+  late int i;
+  do {
+    (i,) = (42,);
+  } while (i < 0);
+}
+
+test3() {
+  late int i;
+  do {
+    (x: i) = (x: 42);
+  } while (i < 0);
+}
+
+test4() {
+  late int i;
+  do {
+    (i,) = (42,);
```

Comment:
Should this be an object pattern based pattern assignment?

Reply:
Of course! Thank you for noticing! Fixed.